src/app/users/request-phone-code.ts

```diff
@@ -0,0 +1,138 @@
+import {
+  getGaloyInstanceName,
+  getRequestPhoneCodeIpLimits,
+  getRequestPhoneCodeLimits,
+  getRequestPhoneCodeMinInterval,
+} from "@config/app"
+import { randomIntFromInterval } from "@core/utils"
+import { UnknownPhoneProviderServiceError } from "@domain/errors"
+import { GeetestError } from "@domain/geetest/error"
+import { RateLimitPrefix } from "@domain/rate-limit"
+import {
+  RateLimiterExceededError,
+  UserPhoneCodeAttemptIpRateLimiterExceededError,
+  UserPhoneCodeAttemptMinIntervalLimiterExceededError,
+  UserPhoneCodeAttemptPhoneRateLimiterExceededError,
+} from "@domain/rate-limit/errors"
+import { GeeTestType } from "@services/geetest.types"
+import { PhoneCodesRepository } from "@services/mongoose/phone-code"
+import { RedisRateLimitService } from "@services/rate-limit"
+import { TwilioClient } from "@services/twilio"
+import { isTestAccountPhone } from "."
+
+export const requestPhoneCodeWithCaptcha = async ({
+  phone,
+  geetest,
+  geetestChallenge,
```

Comment:
I see `geetest` as a service implementation - not sure these fields should be coupled to it.

Reply:
I have not idea how to interpret those fields / give them a generic name. I don't think we should assume captcha would easily be interchangeable, they're meant to be a black box and I'm sure they would have different input/outflow if we choose to add another one